### PR TITLE
Confidential unblind utxo

### LINKED ISSUE #60: [Confidential] Nonce hash
We need to expose a function`NonceHash` that returns the sha256 hash of the ecdh point calculated for the given key pair (public, private keys).

The [`nonceHash`](https://github.com/vulpemventures/liquidjs-lib/blob/97673986c4233dee61f98dc59b5256ed2a8f313d/ts_src/confidential.ts#L6) function of the *confidential* module from liquidjs-lib can be used as reference implementation.

### LINKED ISSUE #61: [Confidential] Unblind utxo
Unblinding a utxo basically means exposing a `UnblindOutput` method that calls the binding for the `secp256k1_rangeproof_rewind` function by providing a private blinding key along with the utxo's confidential fields. 

The [`unblindOutput`](https://github.com/vulpemventures/liquidjs-lib/blob/97673986c4233dee61f98dc59b5256ed2a8f313d/ts_src/confidential.ts#L52) function of the *confidential* module of the liquidjs library could be used as reference implementation for this.

NOTE:
* Before calling the *rangeproof_rewind* binding, the private blinding key and the public one (a.k.a. utxo nonce) should be used as described in #60.
* You can see [here](https://github.com/vulpemventures/liquidjs-lib/blob/97673986c4233dee61f98dc59b5256ed2a8f313d/ts_src/psbt.ts#L711) how this method is used by the "blinder role" of the js version.
* Instead of parsing the generator (a.k.a. asset commitment) with the binding, we have a useful factory method `GeneratorFromBytes` that already does this.

### PR DESCRIPTION
**Pull request by sekulicd**
Before there was no option to unblind confidential transaction output.

After the method UnblindOutput is added, the user can unblind confidential transaction and see real values of sent amount and asset by using the confidential package.

It closes #60.
It closes #61.

Please @altafan can you review this?

### CODE REVIEW

File: transaction/confidential.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package transaction
```
Comment (altafan) [inline transaction/confidential.go:1]: This should not stay in the `transaction` package, but rather we should create a dedicated `confidential` one.

File: transaction/confidential.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+type UnblindOutputResult struct {
+	Value               uint64
+	Asset               []byte
+	ValueBlindingFactor []byte
+	AssetBlindingFactor []byte
+}
+
+//NonceHash method generates hashed secret based on ecdh
+func NonceHash(ctx *secp256k1.Context, pubKey *secp256k1.PublicKey, privKey []byte) (*[32]byte, error) {
```
[the remark this answers is not in the document]
Comment (sekulicd) [reply transaction/confidential.go]: done

File: transaction/confidential.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	result := sha256.Sum256(ecdh)
+	return &result, nil
+}
+
+//UnblindOutput method unblinds confidential transaction output
+func UnblindOutput(input UnblindInput) (*UnblindOutputResult, error) {
+	ctx, _ := secp256k1.ContextCreate(secp256k1.ContextBoth)
+	defer secp256k1.ContextDestroy(ctx)
+
+	_, pubKey, err := secp256k1.EcPubkeyParse(ctx, input.EphemeralPubkey)
```
[the remark this answers is not in the document]
Comment (sekulicd) [reply transaction/confidential.go]: done

File: transaction/confidential.go (hunk 1 above)
Comment (sekulicd) [reply transaction/confidential.go]: done

**Pull request merged**